web/api/web_api_v1.c

```diff
@@ -776,14 +776,57 @@ static inline void web_client_api_request_v1_info_summary_alarm_statuses(RRDHOST
 static inline void web_client_api_request_v1_info_mirrored_hosts(BUFFER *wb) {
     RRDHOST *rc;
     int count = 0;
+
+    buffer_strcat(wb, "\t\"mirrored_hosts\": [\n");
     rrd_rdlock();
     rrdhost_foreach_read(rc) {
         if (rrdhost_flag_check(rc, RRDHOST_FLAG_ARCHIVED))
             continue;
-        if(count > 0) buffer_strcat(wb, ",\n");
+        if (count > 0)
+            buffer_strcat(wb, ",\n");
+
         buffer_sprintf(wb, "\t\t\"%s\"", rc->hostname);
         count++;
     }
+
+    buffer_strcat(wb, "\n\t],\n\t\"mirrored_hosts_status\": [\n");
+    count = 0;
+    rrdhost_foreach_read(rc) {
+        if (rrdhost_flag_check(rc, RRDHOST_FLAG_ARCHIVED))
+            continue;
+        if (count > 0)
+            buffer_strcat(wb, ",\n");
+
```

Comment:
I don't think we should nest those locks, they are completely unrelated and I don't like the implications. We could just copy the strings somewhere or maybe use `buffer_strcat()`?

Reply:
avoided holding both locks at expense of strdupz